lib/net.js

```diff
@@ -834,6 +834,15 @@ function connect(self, address, port, addressType, localAddress, localPort) {
 }
 
 
+// Check that the port number is not NaN when coerced to a number,
+// is an integer and that it falls within the legal range of port numbers.
+function isLegalPort(port) {
+  if (typeof port === 'string' && port.trim() === '')
+    return false;
```

Comment:
Okay, thanks for the explanation.  It feels kind of quirky but I don't have a strong enough opinion to file a PR for it. :-)